libs/MeshKernel/src/CurvilinearGrid/CurvilinearGrid.cpp

```diff
@@ -1304,9 +1310,91 @@ std::vector<CurvilinearGrid::CurvilinearFaceNodeIndices> CurvilinearGrid::Comput
     return result;
 }
 
+std::set<CurvilinearGrid::CurvilinearEdge> CurvilinearGrid::ComputeBoundaryEdges(const CurvilinearGridNodeIndices& lowerLeft, const CurvilinearGridNodeIndices& upperRight) const
+{
+    std::vector<Point> result;
+    const auto facesIndices = ComputeFaceIndices(lowerLeft, upperRight);
+    std::set<CurvilinearEdge> boundaryEdges;
+
+    for (const auto& faceIndices : facesIndices)
+    {
+        if (!std::ranges::all_of(faceIndices, [this](const CurvilinearGridNodeIndices& curviNode)
+                                 { return GetNode(curviNode.m_n, curviNode.m_m).IsValid(); }))
+        {
+            continue;
+        }
+
+        const auto numFaceNodes = static_cast<UInt>(faceIndices.size());
+        for (UInt i = 0u; i < numFaceNodes; ++i)
+        {
+            const auto firstCurvilinearNodeIndex = faceIndices[i];
+            const auto nextIndex = NextCircularForwardIndex(i, numFaceNodes);
+            const auto secondCurvilinearNodeIndex = faceIndices[nextIndex];
+
+            const auto edge = std::make_pair(std::min(firstCurvilinearNodeIndex, secondCurvilinearNodeIndex),
+                                             std::max(firstCurvilinearNodeIndex, secondCurvilinearNodeIndex));
+
+            const auto it = boundaryEdges.find(edge);
```

Comment:
add a comment how this works: while iterating over edges of valid faces, boundary edges are seen once, all internal edges are seen by two faces.